compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -885,3 +903,48 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
         }
     }
 }
+
+pub fn is_trivial_mir(tcx: TyCtxt<'tcx>, did: DefId) -> bool {
```

Comment:
This  could really use a more semantic name. A "trivial" can mean many things, and what it means in this context is not really specified.

Given that its purpose is to check that the body is trivially inlineable, a name for your consideration: `is_trivially_inlineable` or perhaps `is_leaf_mir`.